owncloudApp/src/main/java/com/owncloud/android/shares/datasource/OCRemoteSharesDataSource.kt

```diff
@@ -36,4 +38,23 @@ class OCRemoteSharesDataSource(
     ): RemoteOperationResult<ShareParserResult> {
         return getRemoteSharesForFileOperation.execute(client)
     }
+
+    override fun insertShareForFile(
+        remoteFilePath: String,
+        shareType: ShareType,
+        shareWith: String,
+        permissions: Int,
+        name: String,
+        password: String,
+        expirationTimeInMillis: Long,
+        publicUpload: Boolean,
+        createRemoteShareOperation: CreateRemoteShareOperation
+    ): RemoteOperationResult<ShareParserResult> {
+        createRemoteShareOperation.name = name
+        createRemoteShareOperation.password = password
+        createRemoteShareOperation.expirationDateInMillis = expirationTimeInMillis
+        createRemoteShareOperation.publicUpload = publicUpload
+        createRemoteShareOperation.getShareDetails = true
```

Comment:
Assign something to an attribute that starts with 'get' is not the best name for that attribute... ;-)

Reply:
Done